scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java

```diff
@@ -1074,23 +1128,65 @@ private void build() {
                             }
                             if (name != null && value != null) {
                                 if (!name.matches("[A-Za-z0-9._]+")) {
-                                    // this won't necessarily be a problem at job submission,
-                                    // but it definitely will be here in the client; don't bother
+                                    // this won't necessarily be a problem at
+                                    // job submission,
+                                    // but it definitely will be here in the
+                                    // client; don't bother
                                     continue;
                                 }
 
                                 ret.put(name, new JobVariable(name, value, model));
                             }
                         }
                     } catch (JavaScriptException t) {
-                        // Node.hasAttributes() throws if there are no attributes... (GWT 2.1.0)
+                        // Node.hasAttributes() throws if there are no
+                        // attributes... (GWT 2.1.0)
                     }
                 }
             }
         }
         return ret;
     }
 
+    private Boolean isExecutionCalendarGIDefined(String jobDescriptor) {
```

Comment:
This method is long and has 3 nested levels. Can you break it up into smaller methods.

The file is getting very long. Can you refactor that functionality into new classes so that we stop growing this file, ideally shrink it ;).

Reply:
@tobwiens I will address this point too :) Thanks!